addons/mail/static/src/core/message_model.js

```diff
@@ -171,8 +174,11 @@ export class Message {
 
     get isBodyEmpty() {
         return (
-            !this.body ||
-            ["", "<p></p>", "<p><br></p>", "<p><br/></p>"].includes(this.body.replace(/\s/g, ""))
+            !(this.editDate && this.attachments.length) &&
+            (!this.body ||
+                ["", "<p></p>", "<p><br></p>", "<p><br/></p>"].includes(
+                    this.body.replace(/\s/g, "")
+                ))
```

Comment:
This diff doesn't look correct: `isBodyEmpty` should only have condition on `this.body`.
Shouldn't this be in `isEmpty` instead?

Reply:
We use `isBodyEmpty` condition on showing the message content. Maybe with this diff, the name of function doesn't look quite good, but I think the usage of `isEmpty` is different. Actually with this extra condition, we still have _`edited`_ in the blob, in case of message body is deleted completely and there is a attachment, which visually seems like we have something in the message body.

Reply:
It's safer to make another function `hasTextContent`, so that you distinguish both semantically.

Reply:
For example, in showing of reply to, we don't want to see `(edited)`: we want to inline preview the actual body content of parent message.